themes/cucumber-sb/static/js/site.js

```diff
@@ -91,12 +91,71 @@ function showOnly(language) {
   }
 }
 
+function updateQueryParam(selectedLang){
+  let params = location.search.split("&")
+  if(params.length > 1){
+    params.forEach((element, index) => {
+      if(element.includes("lang")){
+        params[index] = 'lang=' + selectedLang
+      }
+    });
+    return location.search = params.join("&")
+  }else{
+    return location.search = 'lang=' + selectedLang
+  }
+}
+
+function getLangFromUrl(){
+  let params = location.search.split("&")
+  var lang  = ''
+  if(params[0].length > 1){
+    params.forEach((element, index) => {
+      if(element.includes("lang")){
+        lang =  params[index].split("=")[1]
+      }
+    });
+  }
+    return lang
+}
+
 // Activate
 
+var supportedLanguages = [
+  "java",
+  "javascript",
+  "ruby",
+  "kotlin",
+  "scala"
+]
+
+var defaulLanguage = 'java'
+
 ready(function() {
+  var selectedLang = getLangFromUrl();
+  if((selectedLang == '' || selectedLang == null)){
+    if(localStorage.getItem('language') == ''){
+      showOnly(defaulLanguage)
+      localStorage.setItem('language', defaulLanguage)
+    }else{
+      showOnly(localStorage.getItem('language'))
+    }
```

Comment:
Maybe these lines (136-141) which, if I'm reading it correctly, are duplicated down on line 146, could be extracted into a single function to improve readability/maintainability?

Reply:
I want to ship this! Could you consider adding a separate refactoring PR with this nit-pick in it?